src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -48,147 +46,95 @@
  */
 public class ProCombatMoveAI {
 
-  // Utilities
   private final ProAI ai;
-  private final ProTransportUtils transportUtils;
-  private final ProMoveOptionsUtils attackOptionsUtils;
-  private final ProTerritoryValueUtils territoryValueUtils;
-
-  // Current data
   private GameData data;
   private PlayerID player;
-  private Territory myCapital;
+  private ProTerritoryManager territoryManager;
   private boolean isDefensive;
-  private double minCostPerHitPoint;
 
-  public ProCombatMoveAI(final ProAI ai, final ProTransportUtils transportUtils,
-      final ProMoveOptionsUtils attackOptionsUtils, final ProTerritoryValueUtils territoryValueUtils) {
+  public ProCombatMoveAI(final ProAI ai) {
     this.ai = ai;
-    this.transportUtils = transportUtils;
-    this.attackOptionsUtils = attackOptionsUtils;
-    this.territoryValueUtils = territoryValueUtils;
   }
 
-  public Map<Territory, ProTerritory> doCombatMove(final IMoveDelegate moveDel, final GameData data,
-      final PlayerID player, final boolean isSimulation) {
+  public Map<Territory, ProTerritory> doCombatMove(final IMoveDelegate moveDel, final boolean isSimulation) {
     ProLogger.info("Starting combat move phase");
 
     // Current data at the start of combat move
-    this.data = data;
-    this.player = player;
-
-    myCapital = TerritoryAttachment.getFirstOwnedCapitalOrFirstUnownedCapital(player, data);
-
-    // Initialize data containers
-    final Map<Territory, ProTerritory> attackMap = new HashMap<Territory, ProTerritory>();
-    final Map<Unit, Set<Territory>> unitAttackMap = new HashMap<Unit, Set<Territory>>();
-    final Map<Unit, Set<Territory>> transportAttackMap = new HashMap<Unit, Set<Territory>>();
-    final Map<Unit, Set<Territory>> bombardMap = new HashMap<Unit, Set<Territory>>();
-    final List<ProTransport> transportMapList = new ArrayList<ProTransport>();
+    data = ProData.getData();
+    player = ProData.getPlayer();
+    territoryManager = new ProTerritoryManager();
 
     // Determine whether capital is threatened and I should be in a defensive stance
-    isDefensive = !ProBattleUtils.territoryHasLocalLandSuperiority(myCapital, 3, player);
+    isDefensive = !ProBattleUtils.territoryHasLocalLandSuperiority(ProData.myCapital, 3, player);
```

Comment:
magic number code smell "3"

Code would be more clear if this is assigned to a local variable in the prior line, then pass the variable. The variable would serve only as a label. Better yet it would be some public static final constant some where, but at minimum labeling what the value represents is desired.